src/checker.ts

```diff
@@ -400,6 +438,28 @@ export function makeFunctionlessChecker(
     return false;
   }
 
+  /**
+   * Invokes a predicate on the ancestors of the given node until:
+   * * the predicate is true
+   * * there is no parent
+   * * or the limit parameter is hit
+   *
+   * @param limit - An upper bound node to stop processing at.
+   */
+  function findParent<T extends ts.Node>(
+    node: ts.Node,
+    predicate: (node: ts.Node) => node is T,
+    limit?: ts.Node
+  ): T | undefined {
+    if (!node.parent || (limit && limit === node.parent)) {
+      return undefined;
+    } else if (predicate(node.parent)) {
+      return node.parent;
+    } else {
+      return findParent(node.parent, predicate);
+    }
+  }
+
```

Comment:
This doesn't use any aspect of the type checker (only the ast). Does it belong in the checker?

Reply:
There is quite a bit in the checker that does not use the wrapped checker. Did we want to remove them all? 

```ts
  function isStepFunction(node: ts.Node) {
    return (
      isFunctionlessClassOfKind(node, StepFunction.FunctionlessType) ||
      isFunctionlessClassOfKind(node, ExpressStepFunction.FunctionlessType)
    );
  }

  function isApiIntegration(node: ts.Node): node is ts.NewExpression {
    return (
      ts.isNewExpression(node) &&
      isFunctionlessClassOfKind(node.expression, ApiMethod.FunctionlessType)
    );
  }
```

I see some in the checker file, but not in the checker class.

```ts
/**
 * Check if a {@link token} is an {@link ArithmeticToken}: `+`, `-`, `*` or `/`.
 */
export function isArithmeticToken(
  token: ts.SyntaxKind
): token is ArithmeticToken {
  return ArithmeticOperators.includes(token as ArithmeticToken);
}
```

and some in util.ts

```ts
export function hasParent(node: ts.Node, parent: ts.Node): boolean {
  if (!node.parent) {
    return false;
  } else if (node.parent === parent) {
    return true;
  }
  return hasParent(node.parent, parent);
}
```